crates/matrix-sdk/src/widget_api/permissions.rs

```diff
@@ -0,0 +1,24 @@
+//! Types and traits related to the permissions that a widget can request from a client.
+
+use async_trait::async_trait;
+
+/// A trait that must be implemented by an entity that performs the validation / authorization
+/// of the widget's permissions. Typically, it's a native client that "hosts" the widget.
+#[async_trait]
+pub trait PermissionsProvider: Send + Sync + 'static {
+    /// Receives a request for given permissions and returns the actual permissions that
+    /// the user (client) granted to a given widget.
```

Comment:
```suggestion
    /// the clients grants to a given widget (usually by prompting the user).
```

If I'm understanding correctly?

Reply:
Yes, sounds right 🙂